egs/wsj/s5/steps/nnet3/dump_bottleneck_features.sh

```diff
@@ -0,0 +1,147 @@
+   exit 1;
+fi
+
+data=$1
+bnf_data=$2
+nnetdir=$3
+archivedir=$4
+dir=$5
+
+# Assume that final.nnet is in nnetdir
+cmvn_opts=`cat $nnetdir/cmvn_opts`;
+bnf_nnet=$nnetdir/final.raw
+if [ ! -f $bnf_nnet ] ; then
+  echo "No such file $bnf_nnet";
+  exit 1;
+fi
+
+if $use_gpu; then
+  train_queue_opt="--gpu 1"
+  prior_gpu_opt="--use-gpu=yes"
+  if ! cuda-compiled; then
+    echo "$0: WARNING: you are running with one thread but you have not compiled"
+    echo "   for CUDA.  You may be running a setup optimized for GPUs.  If you have"
+    echo "   GPUs and have nvcc installed, go to src/ and do ./configure; make"
+    exit 1
+  fi
+else
+  echo "$0: without using a GPU this will be very slow.  nnet3 does not yet support multiple threads."
+  parallel_train_opts="--use-gpu=no"
+  prior_gpu_opt="--use-gpu=no"
+  prior_queue_opt=""
+fi
+
+
+## Set up input features of nnet
+if [ -z "$feat_type" ]; then
+  if [ -f $nnetdir/final.mat ]; then feat_type=lda; fi
+fi
+echo "$0: feature type is $feat_type"
+
+if [ "$feat_type" == "lda" ] && [ ! -f $nnetdir/final.mat ]; then
+  echo "$0: no such file $nnetdir/final.mat"
+  exit 1
+fi
+
+name=`basename $data`
+sdata=$data/split$nj
+
+mkdir -p $dir/log
+mkdir -p $bnf_data
+echo $nj > $nnetdir/num_jobs
+splice_opts=`cat $nnetdir/splice_opts 2>/dev/null`
+[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj || exit 1;
+
+if [ "$ivector_dir" != "" ];then 
+  use_ivector=true
+fi
+case $feat_type in
+  raw) feats="ark,s,cs:apply-cmvn $cmvn_opts --utt2spk=ark:$sdata/JOB/utt2spk scp:$sdata/JOB/cmvn.scp scp:$sdata/JOB/feats.scp ark:- |"
+       ivec_feats="scp:utils/filter_scp.pl $sdata/JOB/utt2spk $ivector_dir/ivector_online.scp |";;
+  lda) feats="ark,s,cs:apply-cmvn $cmvn_opts --utt2spk=ark:$sdata/JOB/utt2spk scp:$sdata/JOB/cmvn.scp scp:$sdata/JOB/feats.scp ark:- | splice-feats $splice_opts ark:- ark:- | transform-feats $nnetdir/final.mat ark:- ark:- |"
+       ivec_feats="scp:utils/filter_scp.pl $sdata/JOB/utt2spk $ivector_dir/ivector_online.scp |";;
+   
+  *) echo "Invalid feature type $feat_type" && exit 1;
+esac
+
+if [ ! -z "$transform_dir" ]; then
+  echo "Using transforms from $transform_dir"
+  [ ! -f $transform_dir/trans.1 ] && echo "No such file $transform_dir/trans.1" && exit 1;
+  transform_nj=`cat $transform_dir/num_jobs` || exit 1;
+  if [ "$nj" != "$transform_nj" ]; then
+    for n in $(seq $transform_nj); do cat $transform_dir/trans.$n; done >$dir/trans.ark
+    feats="$feats transform-feats --utt2spk=ark:$sdata/JOB/utt2spk ark,s,cs:$dir/trans.ark ark:- ark:- |"
+  else
+    feats="$feats transform-feats --utt2spk=ark:$sdata/JOB/utt2spk ark,s,cs:$transform_dir/trans.JOB ark:- ark:- |"
+  fi
+fi
+
+
+if [ $stage -le 1 ]; then
+  echo "Making BNF scp and ark."
+  echo output-node name=output input=$bnf_name > output.config
+  modified_bnf_nnet="nnet3-copy --rename-node-names=output/output-bkp $bnf_nnet -  | nnet3-init - output.config - |"
```

Comment:
Let's just assume there was no node named output-bkp.  But about this-- please see Issue #1040, eventually it will be possible to rename the node via config files using nnet3-init or nnet-copy.  I'm not sure if we should finish that issue before merging this, but anyway, proceed as if we were merging this first.